propagators/aws/xray/idgenerator.go

```diff
@@ -0,0 +1,74 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package xray
+
+import (
+	crand "crypto/rand"
+	"encoding/binary"
+	"encoding/hex"
+	"math/rand"
+	"strconv"
+	"sync"
+	"time"
+
+	"go.opentelemetry.io/otel/trace"
+)
+
+// IDGenerator is used for generating a new traceID and spanID
+type IDGenerator struct {
+	sync.Mutex
+	randSource *rand.Rand
+}
```

Comment:
Please update to use the new ID generator: https://github.com/open-telemetry/opentelemetry-go/blob/970755bd087801619575b7473806356818e24e15/sdk/trace/id_generator.go#L27-L31